### PR TITLE
Reword and clean up docs

### PR DESCRIPTION
**Pull request by kek**
Started out to fix some typos, ended up rewriting a lot.

### CODE REVIEW

**Comment by MikaelAmborn** [comment]
Nice. Merge it. :+1:

**Comment by martinsvalin** [comment]
Sorry, I made a change to the README before I saw this PR. Our changes were largely orthogonal (and you spotted more typos than me). I rebased your PR, to save you the trouble of a merge conflict.

Looks good! :shipit:!

File: README.md [hunk 1]
```diff
@@ -19,53 +21,89 @@ Or install it yourself as:
...
-`Caerbannog::Queue.push` to store the message before sending it to RabbitMQ, and then by a background
-publisher that uses the `all` method to fetch all pushed messages and send them to RabbitMQ.
 
-If you are using this gem from within a Rails application you can run
+`Caerbannog::Queue` needs to be configured with a message storage class that is an
+ActiveRecord model or works like an ActiveRecord model with regards to the
+methods `.all`, `.create!` and `#destroy`, and has two attributes `name` and
+`payload`.
+
+```
```
Comment (DevL) [inline README.md]:
``````
```ruby
``````

File: README.md [hunk 2]
```diff
@@ -19,53 +21,89 @@ Or install it yourself as:
...
+`MessageQueueMessage`, and an initializer file.
+
+This class will be used to store messages when you call
+`Caerbannog::Queue.push`, before they are sent to RabbitMQ,
+
+Call the `Caerbannog::Queue.push` method with two parameters: the message name,
+that will be used as the routing key in RabbitMQ, and the message payload,
+which should be a hash or an array.
+
+```
```
Comment (DevL) [inline README.md]:
``````
```ruby
``````

File: README.md [hunk 3]
```diff
@@ -19,53 +21,89 @@ Or install it yourself as:
...
+    puts "Oh no" # Handle the error somehow
+    sleep 10 # Wait for RabbitMQ to come back up
+    retry # Try to reconnect
   end
 end
 ```
 
+and you might run this process like
+
+```
```
Comment (DevL) [inline README.md]:
``````
```ruby
``````

File: README.md [hunk 4]
```diff
@@ -19,53 +21,89 @@ Or install it yourself as:
...
+bundle exec rails runner 'MessageQueueWorker'
+```
```
Comment (DevL) [inline README.md]:
``````
```ruby
``````

**Comment by DevL** [comment]
:cake:

File: README.md (hunk 4 above)
Comment (martinsvalin) [inline README.md]: that's shell, you know. ;)

Comment (DevL) [inline README.md]: oops

``````
```bash
``````

then ;)

Comment (DevL) [inline README.md]: Bah, ignore me. :)

**Pull request merged**